src/Text/Pandoc/Readers/Textile.hs

```diff
@@ -535,11 +536,17 @@ link = try $ do
 image :: Parser [Char] ParserState Inlines
 image = try $ do
   char '!' >> notFollowedBy space
-  _ <- attributes -- ignore for now, until we have image attributes
+  (ident, cls, kvs) <- attributes
+  let getAtt k styles = case pickStyleAttrProps [k] styles of
+                          Just v  -> [(k, v)]
+                          Nothing -> []
+  let attr = case lookup "style" kvs of
+               Just stls -> (ident, cls, getAtt "width" stls ++ getAtt "height" stls)
```

Comment:
I did some [reading](https://wiki.haskell.org/GHC_optimisations#Common_subexpression_elimination) and testing and now think you're actually right. GHC isn't smart enough.

    $ ghci src/Text/Pandoc/CSS.hs
    *Text.Pandoc.CSS> :set +s

    *Text.Pandoc.CSS> let s = concat $ take 10000 $ repeat "height: 40px; width:50%; color: red; background: blue;"

    *Text.Pandoc.CSS> pickStyleAttrProps ["color"] s
    Just "red"
    (0.46 secs, 454,666,920 bytes)

    *Text.Pandoc.CSS> (pickStyleAttrProps ["color"] s, pickStyleAttrProps ["color"] s)
    (Just "red",Just "red")
    (1.05 secs, 908,294,712 bytes)